From jacek.anaszewski@gmail.com  Thu Mar 22 11:57:33 2018
From: Jacek Anaszewski <jacek.anaszewski@gmail.com>
Date: Mon, 19 Mar 2018 20:23:19 +0100
Subject: led: core: Clear LED_BLINK_SW flag in led_blink_set()
To: stable@vger.kernel.org, gregkh@linuxfoundation.org
Cc: linux-kernel@vger.kernel.org, linux-leds@vger.kernel.org, pavel@ucw.cz, jacek.anaszewski@gmail.com, Matthieu CASTET <matthieu.castet@parrot.com>
Message-ID: <1521487399-10447-3-git-send-email-jacek.anaszewski@gmail.com>

From: Jacek Anaszewski <jacek.anaszewski@gmail.com>

[Only needed in 4.9.y due to other fixes in mainline - gregkh]

With the current code, the following sequence won't work :
echo timer > trigger

echo 0 >  delay_off
* at this point we call
** led_delay_off_store
** led_blink_set
---
 drivers/leds/led-core.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/leds/led-core.c
+++ b/drivers/leds/led-core.c
@@ -188,6 +188,7 @@ void led_blink_set(struct led_classdev *
 {
 	del_timer_sync(&led_cdev->blink_timer);
 
+	led_cdev->flags &= ~LED_BLINK_SW;
 	led_cdev->flags &= ~LED_BLINK_ONESHOT;
 	led_cdev->flags &= ~LED_BLINK_ONESHOT_STOP;
 
